src/Data/Type/Rec.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Data.Type.Rec
+  ( type Rec,
+    constMap,
+    setRec,
+    getRec,
+    pattern RCon,
+    RecT (RConT, RNil),
+    fieldMap,
+    constMapT,
+    fields,
+    type (.|),
+    HasSel (..),
+
+    -- * Re-exports
+    Ascribe,
+    type (#),
+    Name (Name),
+  )
+where
+
+import Control.Natural (type (~>))
```

Comment:
Unless you're using other things from `natural-transformation`, you could just define your own `~>` type operator to avoid the extra dependency.